src/openvino-ir.js

```diff
@@ -0,0 +1,656 @@
+                }
+            }
+        }
+        return true;
+    }
+}
+
+openvinoIR.DotModel = class {
+    constructor(netDef, init) {
+        var graph = new openvinoIR.DotGraph(netDef, init);
+        this._graphs = [graph];
+    }
+
+    get format() {
+        return 'OpenVINO IR Dot';
+    }
+
+    get graphs() {
+        return this._graphs;
+    }
+}
+
+openvinoIR.DotGraph = class {
+    constructor(netDef, init) {
+        this._name = netDef.id || '';
+        this._version = Boolean(netDef.strict).toString();
+
+        this._nodes = [];
+        this._operators = {};
+        this._inputs = [];
+        this._outputs = [];
+
+        const layers = netDef.children.filter((child) => child.type === "node_stmt");
+        const edges = netDef.children.filter((child) => child.type === "edge_stmt");
+
+        layers.forEach((layer) => {
+            const node = new openvinoIR.DotNode(layer, this._version, edges, layers);
+            this._operators[node.operator] = this._operators[node.operator] ? this._operators[node.operator] + 1 : 1;
+            this._nodes.push(node);
+        });
+
+        edges.forEach((edge) => {
+            const from = edge.edge_list[0];
+            const to = edge.edge_list[1];
+            const child = this._nodes.find((node) => node._id === to.id);
+            if (child) {
+                child.updateInputs(from.id);
+            }
+            const parent = this._nodes.find((node) => node._id === from.id);
+            if (parent) {
+                parent.updateOutputs(to.id);
+            }
+        });
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get inputs() {
+        return this._inputs;
+    }
+
+    get outputs() {
+        return this._outputs;
+    }
+
+    get nodes() {
+        return this._nodes;
+    }
+
+    get operators() {
+        return this._operators;
+    }
+}
+
+openvinoIR.DotNode = class extends openvinoIR.AbstractNode {
+    constructor(layer, version, edges, layers) {
+        super();
+        this._inputs = [];
+        this._outputs = [];
+        this._id = layer.node_id.id;
+
+        this._initializers = [];
+        this._attributes = [];
+
+        const ownAttributes = ['name', 'shape', 'style', 'fillcolor', 'type'];
+
+        layer.attr_list.forEach(({name, value}) => {
+            name = name.toLowerCase().replace(/\s/g, '_');
+            if (ownAttributes.includes(name)) {
+                this[`_${name}`] = value;
+            }
+
+            this._attributes.push(new openvinoIR.Attribute(this, name, value));
+        });
+
+        if (!this._type){
+            this._type = 'data';
+        }
+    }
+
+    updateInputs(id) {
+        this._inputs.push(id);
+    }
+
+    updateOutputs(id) {
+        this._outputs.push(id);
+    }
+}
+
+if (typeof module !== 'undefined' && typeof module.exports === 'object') {
+    module.exports.ModelFactory = openvinoIR.ModelFactory;
```

Comment:
`module.exports.ModelFactory = openvino.ModelFactory;`

Reply:
renamed

is the discussion resolved now?